ls1intum/Artemis
# `Exam mode`: Add checklist overview

### Checklist
#### General


#### Server
#### Client

### Motivation and Context


This PR introduces a smaller version of the exam checklist as overview on exam management page

### Description

We decided to replace the current table columns representing a wide range of different dates with some kind of roadmap summarizing the associated exam checklist.
This roadmap is separated in three steps:

1. Exam preparation
2. Exam conduction
3. Exam correction

Each step contains some sub steps that are displayed below its title.

### Steps for Testing

Testing this is quite tricky as we need to check several different states a exam could have.
Basically, every sub step in exam preparation is something that can be sufficiently done or not. Depending on this, it is shown a green check or a red cross next to the sub step.
Below I added a table containing every factor that leads for a specific sub step of exam preparation to be done sufficiently or not. In this table, the different states for the bubbles above the different steps are listed as well.

**Important**: Also check that there are no inconsistencies between the overview and the actual exam checklist (especially when it comes to the amount of submitted and started exams or the complaints)
Prerequisites:
- 1 Instructor
- Several Exams with different states

0. For each possible state, bring an exam to this state, then:
1. Log in to Artemis
2. Navigate to Course Administration
3. Navigate to exam management tab of the course and inspect the overview
4. Check for inconsistencies between overview and exam checklist

![image](https://user-images.githubusercontent.com/80622272/151666020-64f70898-e53b-4c19-b868-dad01d0d5717.png)

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage





`exam-status.component.ts`: 97.56% | 100%
`exam-checklist.component.ts`: 75% | 100%
`exam-checklist.service.ts`: 84.84% | 100%


### Screenshots


![image](https://user-images.githubusercontent.com/80622272/151666863-aa401b58-0e40-4ec2-8ad1-8ff6f48f046a.png)
